internal/consoleplugin/config.go

```diff
@@ -0,0 +1,205 @@
+package consoleplugin
+
+import (
+	"fmt"
+
+	consolev1alpha1 "github.com/openshift/api/console/v1alpha1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/factory"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Config holds configuration for the logging console plugin.
+// Construct with NewConfig.
+type Config struct {
+	CreatedBy   string
+	Name        string
+	Namespace   string
+	Image       string
+	LokiService string
+	LokiPort    int32
+
+	nginxPort, defaultMode, replicas int32
+}
+
+// NewConfig returns a config with default settings.
+// Modify exported fields to change settings.
+func NewConfig(namespace, createdBy, lokiService string) *Config {
+	return &Config{
+		CreatedBy:   createdBy,
+		Name:        "logging-view-plugin",
+		Namespace:   namespace,
+		Image:       utils.GetComponentImage(constants.ConsolePluginName),
+		LokiService: lokiService,
+		LokiPort:    3001,
+
+		nginxPort:   9443,
+		defaultMode: 420,
+		replicas:    1,
+	}
+}
+
+func (cf *Config) secret() string              { return cf.Name + "-secret" }
+func (cf *Config) configMapName() string       { return cf.Name + "-conf" }
+func (cf *Config) selector() map[string]string { return map[string]string{"app": cf.Name} }
+
+// setup adds common settings for all objects
+func (cf *Config) setup(o client.Object) {
+	l := o.GetLabels()
+	if l == nil {
+		l = map[string]string{}
+	}
+	l[constants.LabelApp] = cf.Name
+	l[constants.LabelK8sName] = cf.Name
+	l[constants.LabelK8sCreatedBy] = cf.CreatedBy
+	o.SetLabels(l)
+}
+
+func (cf *Config) configMap() *corev1.ConfigMap {
+	o := runtime.NewConfigMap(cf.Namespace, cf.configMapName(),
+		map[string]string{
+			"nginx.conf": fmt.Sprintf(`
+				error_log /dev/stdout info;
+				events {}
+				http {
+					access_log         /dev/stdout;
+					include            /etc/nginx/mime.types;
+					default_type       application/octet-stream;
+					keepalive_timeout  65;
+					server {
+						listen              %v ssl;
+						ssl_certificate     /var/serving-cert/tls.crt;
+						ssl_certificate_key /var/serving-cert/tls.key;
+						root                /usr/share/nginx/html;
+					}
+				}
```

Comment:
Does it make sense to have this as a "file embed" instead of a string constant?

Reply:
??

Reply:
Maybe, but it's pretty short and this is simple. A separate file is one more thing to keep track of.